t/t2800-jobs-cmd.t

```diff
@@ -27,83 +27,131 @@ runpty="${SHARNESS_TEST_SRCDIR}/scripts/runpty.py -f asciicast"
 
+listjobs() {
+	${FLUX_BUILD_DIR}/t/job-manager/list-jobs | awk '{print $1}'
+}
+
 test_expect_success 'submit jobs for job list testing' '
-        for i in `seq 1 4`; do \
-            jobid=`flux mini submit hostname`; \
-            flux job wait-event $jobid clean; \
-            echo $jobid >> job_ids1.out; \
-        done &&
-        jobid=`flux mini submit nosuchcommand` &&
-        flux job wait-event $jobid clean &&
-        echo $jobid >> job_ids1.out &&
-        for i in `seq 1 8`; do \
-            jobid=`flux mini submit --time-limit=5m sleep 600`; \
-            flux job wait-event $jobid start; \
-            echo $jobid >> job_ids2.out; \
-        done &&
-        tac job_ids2.out > job_ids_running.out &&
-        flux mini submit --priority=30 sleep 600 >> job_ids_pending.out &&
-        flux mini submit --priority=25 sleep 600 >> job_ids_pending.out &&
-        flux mini submit --priority=20 sleep 600 >> job_ids_pending.out &&
-        flux mini submit --priority=15 sleep 600 >> job_ids_pending.out &&
-        flux mini submit --priority=10 sleep 600 >> job_ids_pending.out &&
-        flux mini submit --priority=5 sleep 600 >> job_ids_pending.out &&
-        jobid=`flux mini submit cancelledjob` &&
-        flux job wait-event $jobid depend &&
-        flux job cancel $jobid &&
-        flux job wait-event $jobid clean &&
-        echo $jobid >> job_ids1.out &&
-        tac job_ids1.out > job_ids_inactive.out &&
-        wait_states
+	#  Create `hostname` and `sleep 600` jobspec
+	#
+	flux mini submit --dry-run hostname >hostname.json &&
+	flux mini submit --dry-run --time-limit=5m sleep 600 > sleep600.json &&
+	#
+	#  Stop the queue so that job-manager can be queried for correct job
+	#   queue order
+	#
+	flux queue stop &&
+	for i in $(seq 0 3); do
+		flux job submit hostname.json >> inactive.ids
+	done &&
+	#
+	#  Currently all inactive ids are "completed"
+	#
+	cp inactive.ids completed.ids &&
+	#
+	#  Start queue and wait for all jobs to finish
+	#
+	flux queue start &&
+	flux queue idle &&
+	#
+	#  Run a job that will fail, copy its JOBID to both inactive and
+	#   failed lists.
+	#
+	jobid=`flux mini submit nosuchcommand` &&
+	flux job wait-event $jobid clean &&
+	echo $jobid >> inactive.ids &&
+	echo $jobid > failed.ids &&
+	#
+	#  Submit 8 sleep jobs to fill up resources
+	#
+	for i in $(seq 0 7); do
+		flux job submit sleep600.json >> runids
+	done &&
+	tac runids > run.ids &&
+	#
+	#  Submit a set of jobs with non-default priorities
+	#  N.B. Use `flux job submit` for efficiency
+	#
+	for p in 30 25 20 15 10 5; do
+		flux job submit --priority=$p sleep600.json >> sched.ids
+	done &&
+	listjobs > active.ids &&
+	#
+	#  Submit a job and cancel it
+	#
+	jobid=`flux mini submit cancelledjob` &&
+	flux job wait-event $jobid depend &&
+	flux job cancel $jobid &&
+	flux job wait-event $jobid clean &&
+	echo $jobid >> inactive.ids &&
+	echo $jobid > cancelled.ids &&
+	cat inactive.ids active.ids >> all.ids &&
```

Comment:
as I'm cleaning up `t2230-job-info-list`, noticed that you don't use `all.ids` really anywhere.  And `all.ids` doesn't seem to contain pending or running ids.

Perhaps construct an `all.ids` that can be used in the future?  I'm doing

```
        #                                                                                                                                   
        #  Create an all.ids list                                                                                                           
        #                                                                                                                                   
        cat pending.ids > all.ids &&                                                                                                        
        cat running.ids >> all.ids &&                                                                                                       
        cat inactive.ids >> all.ids &&  
```

Reply:
nevermind, I misread your line

Reply:
This went through many many iterations, so I would not be surprised if there are some unused things or inconsistencies.